src/caffeine/allocation_s.f90

```diff
@@ -20,12 +20,10 @@
     integer :: me
     type(c_ptr) :: whole_block
     integer(c_ptrdiff_t) :: block_offset
-    integer(c_size_t) :: handle_size, coarray_size, total_size
+    integer(c_size_t) :: handle_size, total_size
     type(handle_data) :: unused
     type(handle_data), pointer :: unused2(:)
```

Comment:
On a related note, PRIF 0.5 mandates that we rename `handle_data` to `prif_coarray_descriptor`, so it might make sense to make that related change in this same PR (especially given both changes are directly relevant to our expected near-term consumer).